scarlet2/frame.py

```diff
@@ -38,75 +40,92 @@ def pixel_size(self):
             return 1
 
     def get_pixel(self, sky_coord):
-        """Get the pixel coordinate from a world coordinate
+        """Get the sky coordinates from a world coordinate
 
         Parameters
         ----------
-        sky_coord: tuple, array
+        sky_coord: SkyCoord, or list of Skycoords
             Coordinates on the sky
+
+        Returns
+        ---------
+        pixel coordinates in the model frame
         """
-        sky = jnp.asarray(sky_coord, dtype=jnp.float32).reshape(-1, 2)
+        assert self.wcs is not None
+        wcs_ = self.wcs.celestial # only use celestial portion
 
-        if self.wcs is not None:
-            wcs_ = self.wcs.celestial  # only use celestial portion
-            pixel = jnp.array(wcs_.world_to_pixel_values(sky)).reshape(-1, 2)
-            # y/x instead of x/y:
-            pixel = jnp.flip(pixel, axis=-1)
+        if isinstance(sky_coord, list):
+            assert all(isinstance(coord, SkyCoord) for coord in sky_coord)
+            pixel = jnp.asarray([coord.to_pixel(wcs_) for coord in sky_coord],
+                                 dtype="float32")     
```

Comment:
I don't think this is necessary. `SkyCoord` works also with lists of coordinates and should also transform lists of coordinates: https://docs.astropy.org/en/stable/api/astropy.coordinates.SkyCoord.html